### PR TITLE
ECI-305 Send Viewed Product Events via client.js

### PR DESCRIPTION
**Pull request by iantance**
Addresses: [ECI-305](https://dripcom.atlassian.net/browse/ECI-305)

### CODE REVIEW

**Review by atongen — APPROVED** [review]
Wow, that was fast! :+1:

File: src/class-drip-woocommerce-view-events.php [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+            currency: "{$currency}",
+            product_url: "{$product->get_permalink()}",
+            image_url: "{$image_url}",
+            source: "drip_woocommerce"
+        }]);
+        </script>
+        <!-- end Drip track -->
+TAG;
+
+      echo $script_tag;
```
Comment (tdieterich-drip) [inline src/class-drip-woocommerce-view-events.php]: So the [wordpress codex](https://codex.wordpress.org/Writing_a_Plugin#Plugin_DevelopmentSuggestions) suggests not doing this, and that the preferred mechanism is to use `wp_enqueue_script(...)`

File: src/class-drip-woocommerce-view-events.php [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    }
+
+    private function product_categories($product) {
+        $categories = array();
+        foreach($product->get_category_ids() as $cid) {
+            $woo_category = get_term_by( 'id', absint( $cid ), 'product_cat' );
+            array_push($categories, $woo_category->name);
```
Comment (will-in-wi) [inline src/class-drip-woocommerce-view-events.php:42]: Nit: A slightly nicer syntax (IMO) is:
```suggestion
            $categories[] = $woo_category->name;
```

Assigning to an empty index in PHP appends to the end of an array.

File: src/class-drip-woocommerce-view-events.php [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        <!-- end Drip track -->
+TAG;
+
+      echo $script_tag;
+    }
+
+    private function product_categories($product) {
+        $categories = array();
+        foreach($product->get_category_ids() as $cid) {
+            $woo_category = get_term_by( 'id', absint( $cid ), 'product_cat' );
```
Comment (will-in-wi) [inline src/class-drip-woocommerce-view-events.php:41]: I'm kind of curious about the `absint` situation. Under what circumstances will the category id of a product be negative?

**Review by will-in-wi — APPROVED** [review]

File: src/class-drip-woocommerce-view-events.php (hunk 3 above)
Comment (iantance) [reply src/class-drip-woocommerce-view-events.php:41]: That method was shameless copied from `class-drip-woocommerce-cart-events.php`, and I didn't look closely at the details. @tdieterich-drip do you know if a category id can be negative?

File: src/class-drip-woocommerce-view-events.php (hunk 1 above)
Comment (iantance) [reply src/class-drip-woocommerce-view-events.php]: 👍 Updated to use `wp_enqueue_script `.

File: src/class-drip-woocommerce-view-events.php (hunk 3 above)
Comment (tdieterich-drip) [reply src/class-drip-woocommerce-view-events.php:41]: That bit of code was shamelessly copied from some code in the woocommerce repository (iirc).  I suspect it's probably not to guard against negative numbers, but a way to make sure we're always dealing with an int no matter what get passed back from get_category_ids().

**Comment by iantance** [comment]
Tested successfully:
Screen Shot 2020-02-26 at 11 31 05 AM

**Review by tdieterich-drip — APPROVED** [review]

**Pull request merged**